target-app/Managers/LocationManager.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  LocationManager.swift
+//  target-app
+//
+//  Created by Andres Leonel Torrico Cossio on 09/09/2022.
+//
+
+import UIKit
+import CoreLocation
+
+final class LocationManager: NSObject, CLLocationManagerDelegate {
+
+    static let shared = LocationManager()
+    private var locationManager: CLLocationManager = CLLocationManager()
+    private var requestLocationAuthorizationCallback: ((CLAuthorizationStatus) -> Void)?
+
+    public func requestLocationAuthorization() {
+        self.locationManager.delegate = self
+        let currentStatus = locationManager.authorizationStatus
+
+        // Only ask authorization if it was never asked before
+        guard currentStatus == .notDetermined else { return }
+
+        // Starting on iOS 13.4.0, to get .authorizedAlways permission, you need to
+        // first ask for WhenInUse permission, then ask for Always permission to
+        // get to a second system alert
+        if #available(iOS 13.4, *) {
+            self.requestLocationAuthorizationCallback = { status in
+                if status == .authorizedWhenInUse {
+                    self.locationManager.requestAlwaysAuthorization()
+                }
+            }
+            self.locationManager.requestWhenInUseAuthorization()
```

Comment:
pls use a weak instance of self in the closure, so you don't create a retain cycle.